Makefile

```diff
@@ -96,6 +96,9 @@ nuke: clean devnet-clean
 .PHONY: nuke
 
 devnet-up:
+	@if [ ! -e op-program/bin ]; then \
```

Comment:
Yes and no.  The `cannon-prestate` target has dependencies is `.PHONY`, which means that it's going to be executed every time, regardless of whether it's already been executed.  So, if `cannon-prestate` were added as a dependency, it would end up getting executed every time, which the `if` tries to avoid.  If you look 3 lines below, this is the same pattern that they used for the `devnet-allocs`, so I simply copied it to here.

Really, I think the project's using `make` wrong.  If you're wanting to use `.PHONY` targets, then you should be relying on the underlying tooling (like say `go build` or `go test`) to cache the results and invalidate things as needed.  If on the other hand you're trying to get `make` to only perform steps which are necessary, the targets should not be `.PHONY` and instead, they should specify patterns to look for file changes that impact the output.  The fact that there's go, typescript, javascript, solidity, python, and other stuff all mixed together doesn't make things any easier to reason about.

But, regardless, this change is upstream, if we'd like to try to restructure the Makefile to remove the `.PHONY` targets and to properly detect dependency changes and rebuild as necessary, I'd suggest we tackle this as another PR.